libs/fixture-generators/src/generate-election-package/generate_election_package.ts

```diff
@@ -0,0 +1,229 @@
+
+  zip.file(
+    ElectionPackageFileName.APP_STRINGS,
+    JSON.stringify(appStrings, null, 2),
+    { date: FIXTURES_FILE_DATE }
+  );
+
+  const ballotStrings = mergeUiStrings(electionStrings, hmpbStrings);
+  const electionWithBallotStrings: Election = {
+    ...election,
+    ballotStrings,
+  };
+
+  const electionDefinition = await generateElectionDefinitionForHmpbs(
+    renderer,
+    electionWithBallotStrings
+  );
+
+  zip.file(ElectionPackageFileName.ELECTION, electionDefinition.electionData, {
+    date: FIXTURES_FILE_DATE,
+  });
+  await renderer.cleanup();
+
+  zip.file(
+    ElectionPackageFileName.SYSTEM_SETTINGS,
+    JSON.stringify(DEFAULT_SYSTEM_SETTINGS, null, 2),
+    { date: FIXTURES_FILE_DATE }
+  );
+
+  // If a prior election package was provided, check to see if the app strings and ballot strings / election definition
+  // have changed. If they have not we can reuse the audio clips and audio ids from the prior election package. Otherwise
+  // we will regenerate them.
+  if (
+    priorElectionPackage &&
+    electionDefinition.electionData ===
+      priorElectionPackage.electionDefinition.electionData &&
+    electionDefinition.ballotHash ===
+      priorElectionPackage.electionDefinition.ballotHash &&
+    JSON.stringify(mergeUiStrings(appStrings, ballotStrings)) ===
+      JSON.stringify(priorElectionPackage.uiStrings) &&
+    priorElectionPackage.uiStringAudioIds &&
+    priorElectionPackage.uiStringAudioClips
+  ) {
+    zip.file(
+      ElectionPackageFileName.AUDIO_IDS,
+      JSON.stringify(priorElectionPackage.uiStringAudioIds, null, 2),
+      { date: FIXTURES_FILE_DATE }
+    );
+    zip.file(
+      ElectionPackageFileName.AUDIO_CLIPS,
+      Readable.from(
+        priorElectionPackage.uiStringAudioClips.map(
+          (clip) => `${JSON.stringify(clip)}\n`
+        )
+      ),
+      { date: FIXTURES_FILE_DATE }
+    );
+  } else {
+    const speechSynthesizer = new GoogleCloudSpeechSynthesizerWithoutCache({});
+
+    const { uiStringAudioIds, uiStringAudioClips } = generateAudioIdsAndClips({
+      appStrings,
+      electionStrings,
+      speechSynthesizer,
+    });
+    zip.file(
+      ElectionPackageFileName.AUDIO_IDS,
+      JSON.stringify(uiStringAudioIds, null, 2),
+      { date: FIXTURES_FILE_DATE }
+    );
+    zip.file(ElectionPackageFileName.AUDIO_CLIPS, uiStringAudioClips, {
+      date: FIXTURES_FILE_DATE,
+    });
+  }
+
+  const zipContents = await zip.generateAsync({
+    type: 'nodebuffer',
+    streamFiles: true,
+  });
+  const electionPackageHash = sha256(zipContents);
+  const combinedHash = formatElectionHashes(
+    electionDefinition.ballotHash,
+    electionPackageHash
+  );
+  const fileName = `election-package-${combinedHash}.zip`;
+  const packageFilePath = path.join(assetDirectoryPath, fileName);
+  const electionFilePath = path.join(
+    assetDirectoryPath,
+    `generated-election-${combinedHash}.json`
```

Comment:
> Should the hashes be in the generated election-package and election.json
filenames? This means you have to manually update the fixtures index.ts file
if this changes which is a little annoying, but people should be aware when
they are causing it to change. Maybe we could expose it as a variable more
easily so that tests currently referencing these hardcoded values can at least
be fixed and make this the ONLY place you need to update. I probably will want
to keep these files in a subfolder to make deletion easier instead of what I'm
doing right now.

Hmm, trying to think through the pros/cons to having hashes in the name. I guess the pro is that you can use it to distinguish which package is which if you are using multiple. That feels like it could come in handy. Is there some hack we could use like a symlink or something so that the index.ts file can always load the package even if the file name changes? I.e. when we generate the package we also create a symlink from `election-package.zip` to `election-package-{hash}.zip`?

I like the idea of exposing the hash in a variable as well. Though I think we already have code to do that when we parse election packages, so maybe any test that uses it will just get it through that?